packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -268,9 +268,22 @@ module.exports = (
       if (cachedHeadings) {
         return cachedHeadings
       } else {
+        const getHeadingID = heading => {
+          const data = heading.data
+          if (data) {
+            if (data.id) return data.id
+            if (data.htmlAttributes && data.htmlAttributes.id) {
+              return data.htmlAttributes.id
+            }
+          }
+
+          return ``
```

Comment:
I think this should be `null` and not empty string? Empty string is still a value, `null` should signify lack of it